pulsar-client/src/main/java/org/apache/pulsar/client/impl/RoundRobinPartitionMessageRouterImpl.java

```diff
@@ -50,24 +51,29 @@ public class RoundRobinPartitionMessageRouterImpl extends MessageRouterBase {
 
     private static final Clock SYSTEM_CLOCK = Clock.systemUTC();
 
+    private Supplier<Boolean> batchingCallback;
+
     public RoundRobinPartitionMessageRouterImpl(HashingScheme hashingScheme,
                                                 int startPtnIdx,
                                                 boolean isBatchingEnabled,
-                                                long partitionSwitchMs) {
-        this(hashingScheme, startPtnIdx, isBatchingEnabled, partitionSwitchMs, SYSTEM_CLOCK);
+                                                long partitionSwitchMs,
+                                                Supplier batchingCallback) {
+        this(hashingScheme, startPtnIdx, isBatchingEnabled, partitionSwitchMs, SYSTEM_CLOCK, batchingCallback);
     }
 
     public RoundRobinPartitionMessageRouterImpl(HashingScheme hashingScheme,
                                                 int startPtnIdx,
                                                 boolean isBatchingEnabled,
                                                 long partitionSwitchMs,
-                                                Clock clock) {
+                                                Clock clock,
+                                                Supplier batchingCallback) {
```

Comment:
Nit: use generic type


Also, do we have to remove the isBatchingEnabled field?

What about simply changing the type of isBatchingEnable to Supplier<Boolean>

Reply:
Yeah. I think isBatchingEnabled is useless.